app/src/features/Batches/BatchResourceDialog.tsx

```diff
@@ -0,0 +1,236 @@
+import React, { useState, useEffect, useMemo } from "react";
+
+import {
+  Checkbox,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogTitle,
+  Divider,
+  InputAdornment,
+  List,
+  makeStyles,
+  TextField,
+  Typography,
+} from "@material-ui/core";
+import { Search } from "@material-ui/icons";
+import clsx from "clsx";
+import { useTranslation } from "react-i18next";
+import { useParams } from "react-router";
+
+import Button from "common/components/Button";
+import {
+  useApiBatchesCreateMutation,
+  useApiResourcesListQuery,
+} from "services/api/endpoints";
+
+import BatchResourceListItem from "./BatchResourceListItem";
+
+const useStyles = makeStyles((theme) => ({
+  dialog: {
+    padding: theme.spacing(3),
+    height: 470,
+  },
+  title: {
+    display: "flex",
+    justifyContent: "space-between",
+    alignItems: "center",
+    flexWrap: "wrap",
+  },
+  checkboxForm: {
+    cursor: "pointer",
+  },
+  textField: {
+    padding: theme.spacing(0, 3),
+  },
+}));
+
+type BatchResourceDialogType = {
+  open: boolean;
+  onClose: React.Dispatch<React.SetStateAction<boolean>>;
+  setAlert: React.Dispatch<React.SetStateAction<string | undefined>>;
```

Comment:
```suggestion
  onClose: () => void;
  setAlert: (alert?: string) => void;
```

Reply:
I guess we'll have to rebase after the PR with the Alert component removing is merged.

You're going to have to remove this `setAlert` props